src/ruby/pb/test/client.rb

```diff
@@ -649,6 +649,21 @@ def custom_metadata
 
   end
 
+  def special_status_message
+    code = GRPC::Core::StatusCodes::UNKNOWN
+    message = "\t\ntest with whitespace\r\nand Unicode BMP ☺ and non-BMP 😈\t\n"
+    req = SimpleRequest.new(
+      response_status: EchoStatus.new(code: code, message: message))
+    begin
+      resp = @stub.unary_call(req)
+    rescue GRPC::Unknown => e
+      if e.details.force_encoding("UTF-8") != message
+        fail AssertionError,
+          "Expected message #{message}. Received: #{e.details}"
+      end
+    end
```

Comment:
we should add logic to make sure this test fails if the rescue block is *not* taken

Reply:
Done.

Reply:
This will still "pass" if the call accidentally succeeds, right?

Reply:
Test will fail at line 659 if the call succeeds.
It will fill at the end if the call raises exceptions other than Unknown.